src/phantom.js

```diff
@@ -68,16 +68,16 @@ export default class Phantom {
                 'This generally means something went wrong when installing phantomjs-prebuilt. Exiting.');
         }
 
-        if (typeof logger !== 'object') {
-            throw new Error('logger must be ba valid object.');
+        if (!logger) {
+            throw new Error('logger must be a valid object.');
         }
 
-        logger.debug = logger.debug || (() => undefined);
-        logger.info = logger.info || (() => undefined);
-        logger.warn = logger.warn || (() => undefined);
-        logger.error = logger.error || (() => undefined);
-
-        this.logger = logger;
+        this.logger = {
+            debug: logger.debug && ((...args) => logger.debug(...args)) || (() => undefined),
```

Comment:
I didn't like making changes on the original `logger` object (`logger.debug = ...`) so this is creating a new object (`this.logger = {...}`) taking whatever it needs from it. 

It's also creating a wrapper function for each method so as to preserve `this` (calling `logger.debug` with `logger` as its `this`)